symlink.c

```diff
@@ -80,10 +81,16 @@ static int nova_readlink(struct dentry *dentry, char __user *buffer, int buflen)
 	return readlink_copy(buffer, buflen, blockp);
 }
 
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(4, 5, 0)
+static const char *nova_get_link(struct dentry *dentry, struct inode *inode, void **cookie)
+#else
 static const char *nova_follow_link(struct dentry *dentry, void **cookie)
+#endif
 {
 	struct nova_file_write_entry *entry;
+#if LINUX_VERSION_CODE < KERNEL_VERSION(4, 5, 0)
 	struct inode *inode = dentry->d_inode;
+#endif
```

Comment:
This looks wield. Can you do it this way:

nova_get_link(){
...
}

+#if LINUX_VERSION_CODE < KERNEL_VERSION(4, 5, 0)
+nova_follow_link() {
  ...
  return nova_get_link();
}
+#else